linera-core/src/client/mod.rs

```diff
@@ -2177,7 +2180,10 @@ where
             operations: Vec::new(),
             previous_block_hash,
             height,
-            authenticated_signer: owner,
+            authenticated_signer: owner.and_then(|owner| match owner {
+                AccountOwner::User(user) => Some(user),
+                AccountOwner::Application(_) => None,
```

Comment:
I thought this worked by querying the application balance without using any authenticated signer :thinking: At least I thought that otherwise it wouldn't work for querying the chain balance.

Reply:
Ah, right. Now I understand, thanks! Opened issue #2983 for this.